<?xml version="1.0" encoding="utf-8"?>
<!-- This URDF was automatically created by SolidWorks to URDF Exporter! Originally created by Stephen Brawner (brawner@gmail.com) 
     Commit Version: 1.6.0-4-g7f85cfe  Build Version: 1.6.7995.38578
     For more information, please see http://wiki.ros.org/sw_urdf_exporter -->
<robot
  name="7dofmed.SLDASM">
  <joint name="world" />
  <joint
    name="locate"
    type="fixed">
    <origin
      xyz="0 0 0"
      rpy="0 0 0" />
    <parent
      link="world" />
    <child
      link="base_link" />
    <limit
      lower="0"
      upper="0"
      effort="0"
      velocity="0" />
  </joint>
  <link
    name="base_link">
    <inertial>
      <origin
        xyz="-0.091634 0.3943 1.1864"
        rpy="0 0 0" />
      <mass
        value="8.4839" />
      <inertia
        ixx="0.092378"
        ixy="-4.5235E-19"
        ixz="-1.2365E-18"
        iyy="0.025859"
        iyz="-6.8691E-16"
        izz="0.092378" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="$(find gazebodemo)/meshes/base_link.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.79216 0.81961 0.93333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="$(find gazebodemo)/meshes/base_link.STL" />
      </geometry>
    </collision>
  </link>
  <link
    name="Link1">
    <inertial>
      <origin
        xyz="-0.00032291 -0.00045552 -0.46306"
        rpy="0 0 0" />
      <mass
        value="5.0375" />
      <inertia
        ixx="0.072567"
        ixy="-0.00087892"
        ixz="-6.0449E-05"
        iyy="0.07195"
        iyz="-8.5265E-05"
        izz="0.012642" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="$(find gazebodemo)/meshes/Link1.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0 0 0.75294 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="$(find gazebodemo)/meshes/Link1.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="theta1"
    type="revolute">
    <origin
      xyz="-0.091634 -1.1864 0.80892"
      rpy="-3.1416 0 -0.58752" />
    <parent
      link="base_link" />
    <child
      link="Link1" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="0"
      upper="0"
      effort="0"
      velocity="0" />
  </joint>
  <link
    name="Link2">
    <inertial>
      <origin
        xyz="0.204457133023888 -0.209824454025208 0.344832142709993"
        rpy="0 0 0" />
      <mass
        value="7.93769784757158" />
      <inertia
        ixx="0.0969397568300253"
        ixy="0.0319672198298833"
        ixz="-0.000181281807695844"
        iyy="0.09528284140441"
        iyz="0.000186039742727897"
        izz="0.0762223481731407" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="$(find gazebodemo)/meshes/Link2.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0 0 0.752941176470588 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="$(find gazebodemo)/meshes/Link2.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="theta2"
    type="revolute">
    <origin
      xyz="0 0 -0.71"
      rpy="1.5708 -1.0368 -0.61668" />
    <parent
      link="Link1" />
    <child
      link="Link2" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="0"
      upper="0"
      effort="0"
      velocity="0" />
  </joint>
  <link
    name="Link3">
    <inertial>
      <origin
        xyz="0.00045837 -0.00031885 -0.41472"
        rpy="0 0 0" />
      <mass
        value="5.0375" />
      <inertia
        ixx="0.071935"
        ixy="0.0008733"
        ixz="8.58E-05"
        iyy="0.072582"
        iyz="-5.9687E-05"
        izz="0.012642" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="$(find gazebodemo)/meshes/Link3.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0 0 0.75294 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="$(find gazebodemo)/meshes/Link3.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="theta3"
    type="revolute">
    <origin
      xyz="0.53552 -0.54958 0.435"
      rpy="-1.5708 -0.97511 0.77244" />
    <parent
      link="Link2" />
    <child
      link="Link3" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="0"
      upper="0"
      effort="0"
      velocity="0" />
  </joint>
  <link
    name="Link4">
    <inertial>
      <origin
        xyz="-0.19203 -0.052948 -0.43983"
        rpy="0 0 0" />
      <mass
        value="7.9377" />
      <inertia
        ixx="0.074095"
        ixy="-0.014888"
        ixz="0.00025041"
        iyy="0.12398"
        iyz="6.9047E-05"
        izz="0.070365" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="$(find gazebodemo)/meshes/Link4.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0 0 0.75294 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="$(find gazebodemo)/meshes/Link4.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="theta4"
    type="revolute">
    <origin
      xyz="0 0 -0.66166"
      rpy="-1.5708 -1.1863 0.963" />
    <parent
      link="Link3" />
    <child
      link="Link4" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="0"
      upper="0"
      effort="0"
      velocity="0" />
  </joint>
  <link
    name="Link5">
    <inertial>
      <origin
        xyz="-0.00037583 0.00041294 -0.5127"
        rpy="0 0 0" />
      <mass
        value="5.0375" />
      <inertia
        ixx="0.072346"
        ixy="0.00092733"
        ixz="-7.035E-05"
        iyy="0.072171"
        iyz="7.7299E-05"
        izz="0.012642" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="$(find gazebodemo)/meshes/Link5.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0 0 0.75294 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="$(find gazebodemo)/meshes/Link5.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="theta5"
    type="revolute">
    <origin
      xyz="-0.48622 -0.13406 -0.589"
      rpy="1.5708 0.84837 1.8398" />
    <parent
      link="Link4" />
    <child
      link="Link5" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="0"
      upper="0"
      effort="0"
      velocity="0" />
  </joint>
  <link
    name="Link6">
    <inertial>
      <origin
        xyz="0.083441 0.31569 -0.34691"
        rpy="0 0 0" />
      <mass
        value="8.4097" />
      <inertia
        ixx="0.12861"
        ixy="-0.016042"
        ixz="0.00016753"
        iyy="0.072311"
        iyz="0.00023051"
        izz="0.082846" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="$(find gazebodemo)/meshes/Link6.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.79216 0.81961 0.93333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="$(find gazebodemo)/meshes/Link6.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="theta6"
    type="revolute">
    <origin
      xyz="0 0 -0.75964"
      rpy="-1.5708 0.48779 0.73838" />
    <parent
      link="Link5" />
    <child
      link="Link6" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="0"
      upper="0"
      effort="0"
      velocity="0" />
  </joint>
  <link
    name="Link7">
    <inertial>
      <origin
        xyz="-0.9149 -0.49136 -0.59887"
        rpy="0 0 0" />
      <mass
        value="259.35" />
      <inertia
        ixx="22.351"
        ixy="-8.7163"
        ixz="0.39477"
        iyy="30.438"
        iyz="-0.61783"
        izz="23.11" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="$(find gazebodemo)/meshes/Link7.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0 0 0.75294 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="$(find gazebodemo)/meshes/Link7.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="theta7"
    type="revolute">
    <origin
      xyz="0.17868 0.77298 -0.43067"
      rpy="1.5103 0.40671 -1.8521" />
    <parent
      link="Link6" />
    <child
      link="Link7" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="0"
      upper="0"
      effort="0"
      velocity="0" />
  </joint>
  
  <link name="tip" />
    
  <joint name="attach-tip"
    type="fixed">
    <parent
      link="Link7" />
    <child
      link="tip" />
  <joint />
  
  
  <ros2_control name="GazeboSystem" type="system">
    <hardware>
      <plugin>gazebo_ros2_control/GazeboSystem</plugin>
    </hardware>

    <joint name="theta1">
      <state_interface   name="position"/>
      <state_interface   name="velocity"/>
      <command_interface name="velocity"/>
    </joint>

    <joint name="theta2">
      <state_interface   name="position"/>
      <state_interface   name="velocity"/>
      <command_interface name="velocity"/>
    </joint>

    <joint name="theta3">
      <state_interface   name="position"/>
      <state_interface   name="velocity"/>
      <command_interface name="velocity"/>
    </joint>

    <joint name="theta4">
      <state_interface   name="position"/>
      <state_interface   name="velocity"/>
      <command_interface name="velocity"/>
    </joint>

    <joint name="theta5">
      <state_interface   name="position"/>
      <state_interface   name="velocity"/>
      <command_interface name="velocity"/>
    </joint>

    <joint name="theta6">
      <state_interface   name="position"/>
      <state_interface   name="velocity"/>
      <command_interface name="velocity"/>
    </joint>

    <joint name="theta7">
      <state_interface   name="position"/>
      <state_interface   name="velocity"/>
      <command_interface name="velocity"/>
    </joint>
  </ros2_control>

  <gazebo>
    <plugin name="gazebo_ros2_control" filename="libgazebo_ros2_control.so">
	    <parameters>/home/hope/133ws/src/gazebodemo/config/sevenDOF_control_parameters.yaml</parameters>
    </plugin>
  </gazebo>

</robot>
